lib/multi_configs.js

```diff
@@ -0,0 +1,129 @@
+const fs = require('fs');
+const path = require('path');
+
+const isArray = require('lodash.isarray');
+const isString = require('lodash.isstring');
+const isPlainObject = require('lodash.isplainobject');
+const mapValues = require('lodash.mapvalues');
+
+const expandTilde = require('expand-tilde');
+const parsePath = require('parse-filepath');
+
+
+function search(options, extensions, cwd) {
+  cwd = path.resolve(isString(cwd) ? cwd : '.');
+  extensions = normalizeExtensions(extensions);
+
+  if (!isPlainObject(options)) {
+    return {};
+  }
+
+  return mapValues(options, function(opt, name) {
+    return firstUnNull(expandEachProp(opt, name), findWithExpandedPaths);
+  });
+
+
+  function expandEachProp(opt, name) {
```

Comment:
could the amount of code be reduced by combining all the expand/normalize methods - `expandEachProp` calls `createExpandedPathArray` which calls `normalizeExpandedPath` but I don't think they are called anywhere else.  It might be simpler to combine them and just add logic based on the state of the `opt` (object, string, etc)